containerization-installs/docker-install.sh

```diff
@@ -28,41 +28,91 @@ fi
 # Creating a keyrings directory 
 if (test -d /etc/bin/keyrings)
 then
-  echo "creating keyrings"
+  echo "keyrings directory already exist"
 else
   echo "Creating keyrings"
   sudo install -m 0755 -d /etc/apt/keyrings
 fi
 
+# Check if the keyrings directory exists & if not, create it
+if [ -f /etc/apt/keyrings/docker.asc ]
+then
+  echo "The Docker GPG key already exists"
+else
+  echo "The Docker GPG key does not exist. Downloading it"
+  sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+fi
 
+# Check if the Docker GPG key exists & if not, install it
+if [ -f /etc/apt/keyrings/docker.asc ]
+then
+  echo "The Docker GPG key already exists"
+else
+  echo "The Docker GPG key does not exist. Downloading it"
+  sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+fi
  
 # setting permissions 
-if (test -d /etc/bin/keyrings/docker.asc)
+if (-r /etc/apt/keyrings/docker.asc)
```

Comment:
Idt you meant to delete the `(test ...)` part of this line

Reply:
You are right greg thanks.